pkg/agent/openflow/packetin.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openflow
+
+import (
+	"time"
+
+	"github.com/contiv/ofnet/ofctrl"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+)
+
+type ofpPacketInReason uint
+
+type PacketInHandler interface {
+	HandlePacketIn(pktIn *ofctrl.PacketIn) error
+	GetHandlerName() string
+}
+
+const (
+	// Action explicitly output to controller.
+	ofprAction ofpPacketInReason = 1
+)
+
+func (c *client) RegisterPacketInHandler(packetInHandler interface{}) {
+	handler, ok := packetInHandler.(PacketInHandler)
```

Comment:
This is to avoid generated mock_openflow.go importing openflow package, which cause `import cycle not allowed` error

Reply:
Use PacketInHandler in this interface will cause test case failed by import cycle in mock_openflow.go
```
# github.com/vmware-tanzu/antrea/pkg/agent/openflow
FAIL	github.com/vmware-tanzu/antrea/pkg/agent/openflow [setup failed]
import cycle not allowed in test
package github.com/vmware-tanzu/antrea/pkg/agent/openflow (test)
	imports github.com/vmware-tanzu/antrea/pkg/agent/openflow/testing
	imports github.com/vmware-tanzu/antrea/pkg/agent/openflow
```

Reply:
We could just move interface PacketInHandler to package openflow, probably to client.go?

Reply:
> We could just move interface PacketInHandler to package openflow, probably to client.go?

No, this method will still hit import cycle.
No matter the position of PacketInHandler, this error will occur if we expose PacketInHandler as property of interface:RegisterPacketInHandler, because the auto-generated mock_openflow.go will import openflow package in this case.
Another solution is to make PacketInHandler in a separate package.